sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-kernel/coreos-kernel-6.12.49.ebuild

```diff
@@ -89,6 +101,82 @@ src_compile() {
+	# Copy full initrd over to /usr as filesystem image
+	mkdir "${S}"/build/bootengine || die
+	pushd "${S}"/build/bootengine || die
+	sudo lsinitrd --unpack "${S}"/build/bootengine.cpio || die
+	sudo mksquashfs . "${S}"/build/bootengine.img -noappend -xattrs-exclude ^btrfs. || die
+	popd || die
+	# Create minimal initrd
+	if use amd64; then
+		mkdir "${S}"/build/early-cpio || die
+		pushd "${S}"/build/early-cpio || die
+		sudo lsinitrd --unpackearly "${S}"/build/bootengine.cpio || die
+		# Recreate to only contain the early cpio for microcode
+		find . -print0 | cpio --null --create --verbose --format=newc > "${S}"/build/bootengine.cpio
+		# Debug: List contents after recreation
+		cpio -t < "${S}"/build/bootengine.cpio
+		popd || die
+	else
+		# No early cpio, drop full initrd
+		> "${S}"/build/bootengine.cpio
+	fi
+	mkdir "${S}"/build/minimal || die
+	pushd "${S}"/build/minimal || die
+	mkdir -p {etc,bin,sbin,dev,proc,sys,dev,lib,lib64,usr/bin,usr/sbin,usr/lib,usr/lib64,realinit,sysusr/usr}
+	mkdir -p lib/modules/"${KV_FULL}"/
+	# Instead from ESYSROOT we can also copy kernel modules from the dracut pre-selection
+	cp "${S}"/build/bootengine/usr/lib/modules/"${KV_FULL}"/modules.* lib/modules/"${KV_FULL}"/
+	mkdir -p lib/modprobe.d/
+	cp "${S}"/build/bootengine/lib/modprobe.d/* lib/modprobe.d/
+	# Only include modules related to mounting /usr and for interacting with the emergency console
+	local -a MODULES=("fs/overlayfs" "fs/squashfs" "drivers/md/dm-verity.ko.xz" "drivers/md/dm-mod.ko.xz" "drivers/block/loop.ko.xz" "fs/btrfs" "drivers/nvme" "drivers/scsi" "drivers/ata" "drivers/block" "drivers/pci" "drivers/char/virtio_console.ko.xz" "drivers/hv" "drivers/input/serio" "drivers/mmc" "drivers/usb" "drivers/hid" "security/keys")
+	local MODULE DEP DEP_DIR
+	for MODULE in "${MODULES[@]}"; do
+		if [[ -f ${S}/build/bootengine/usr/lib/modules/${KV_FULL}/kernel/${MODULE} ]]; then
+			local MODULE_DIR=${MODULE%/*}
+			mkdir -p lib/modules/"${KV_FULL}"/kernel/"${MODULE_DIR}"
+			cp "${S}"/build/bootengine/usr/lib/modules/"${KV_FULL}"/kernel/"${MODULE}" lib/modules/"${KV_FULL}"/kernel/"${MODULE}"
+		elif [[ -d ${S}/build/bootengine/usr/lib/modules/${KV_FULL}/kernel/${MODULE} ]]; then
+			mkdir -p lib/modules/"${KV_FULL}"/kernel/"${MODULE}"
+			cp -r "${S}"/build/bootengine/usr/lib/modules/"${KV_FULL}"/kernel/"${MODULE}"/* lib/modules/"${KV_FULL}"/kernel/"${MODULE}"/
+		else
+			die "wrong module type/not found: ${S}"/build/bootengine/usr/lib/modules/"${KV_FULL}"/kernel/"${MODULE}"
+		fi
+	done
+	# Copy module dependencies
+	for MODULE in $(find ./lib/modules/"${KV_FULL}"/ -type f); do
+		MODULE=${MODULE##*/}
+		MODULE=${MODULE%.ko.xz}
+		for DEP in $(modprobe -S "${KV_FULL}" -d "${S}"/build/bootengine -D "${MODULE}" | grep "^insmod " | cut -d " " -f 2); do
+			DEP=${DEP#"${S}/build/bootengine"}
+			DEP_DIR=${DEP%/*}
+			mkdir -p ./"${DEP_DIR}"
+			cp "${S}"/build/bootengine/"${DEP}" ./"${DEP}"
+		done
+	done
```

Comment:
dracut-install can handle modules too, including their dependencies and their firmware! How about this?

```bash
pushd "${S}/build/bootengine/usr/lib/modules/${KV_FULL}" || die
find kernel/drivers/{ata,block,hid,hv,input/serio,mmc,nvme,pci,scsi,usb} kernel/fs/{btrfs,overlayfs,squashfs} kernel/security/keys -name "*.ko.*" -printf "%f\0" | xargs --null "${BROOT}"/usr/lib/dracut/dracut-install --destrootdir "${S}"/build/minimal --kerneldir . --module dm-verity dm-mod virtio_console
popd || die
```

Reply:
Probably want `--firmwaredirs "${ESYSROOT}"/lib/firmware` in there too so that it doesn't take that from the SDK.

Reply:
That helps, I need one more workaround for this broken `--firmwaredirs` flag and then have to test again…

Reply:
If you find something is broken, let me know. I wrote half of dracut-install, but hopefully not the half you're having trouble with. :sweat_smile: Obviously, it doesn't normally get invoked like this.

Reply:
Okay, it _does_ respect `--sysrootdir`, it was just deeper in the logic than I had looked. Passing this should do it:

```
--sysrootdir "${ESYSROOT}" --firmwaredirs "${ESYSROOT}"/lib/firmware
```

I don't know why you chose `"${S}"/build/bootengine/usr/lib/firmware` though?

Reply:
I've tested this out locally to save you waiting to find out. It works. :smile: